lib/sidekiq_adhoc_job/web/routes/jobs/schedule.rb

```diff
@@ -1,15 +1,33 @@
+# frozen_string_literal: true
+
 module SidekiqAdhocJob
   module Web
     module Jobs
       class Schedule
-
         def self.register(app)
           app.post '/adhoc-jobs/:name/schedule' do
-            SidekiqAdhocJob::ScheduleAdhocJob.new(params[:name], request.params).call
+            # Use Sidekiq 8's new API if available, fallback to legacy for Sidekiq 7
```

Comment:
to review

Reply:
Could we check for gem version instead of `respond_to?`